Heading/Heading.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Agate styled labeled Heading components and behaviors
+ *
+ * @example
+ * <Heading
+ *   size="large"
+ *   spacing="small"
+ * >
+ *   A Content Section Heading
+ * </Heading>
+ *
+ * @module agate/Heading
+ * @exports Heading
+ * @exports HeadingBase
+ * @exports HeadingDecorator
+ */
+
+import kind from '@enact/core/kind';
+import Pure from '@enact/ui/internal/Pure';
+import PropTypes from 'prop-types';
+import compose from 'ramda/src/compose';
+import UiHeading from '@enact/ui/Heading';
+
+import Skinnable from '../Skinnable';
+
+import componentCss from './Heading.module.less';
+
+/**
+ * A labeled Heading component.
+ *
+ * This component is most often not used directly but may be composed within another component as it
+ * is within [Heading]{@link agate/Heading.Heading}.
+ *
+ * @class HeadingBase
+ * @memberof agate/Heading
+ * @ui
+ * @public
+ */
+const HeadingBase = kind({
+	name: 'Heading',
+
+	propTypes: /** @lends agate/Heading.HeadingBase.prototype */ {
+		css: PropTypes.object,
+
+		/**
+		 * Set a custom color to be used by the Heading. Defaults to the current accent color if
+		 * none is specified.
+		 *
+		 * @type {String}
+		 * @default null
+		 * @public
+		 */
+		customColor: PropTypes.string,
```

Comment:
Why did you include the "custom" prefix here? That word doesn't carry a lot of meaning for me in this context since any color I'd pass would be "custom". Maybe just `color` or `accentColor` if the purpose of the color needs to be included in the name. Thoughts?

Reply:
I'd originally started with just "color", but I didn't feel that conveyed enough contextual meaning. Implying that without some qualifier, there would be no color. There is in fact a default color, which itself is the accent color. I didn't choose the logical "accentColor" because I didn't want confusion about whether this was overriding/updating the skin's accent color or just applying to this component. "customColor" was the compromise which indicates that you're customizing the color beyond its existing, default color. I can change it back to `color` though, I _think_ I'd be ok with that.

Reply:
I'd be in favor of `color` but `lineColor` might be another option if we support it for the "border" color across sizes.

Reply:
I was wanting to avoid visual terminology so a skin could determine the shape for itself. It's a line in gallium, but it's also slanted. Other skins may leverage this feature differently.

Reply:
I updated it to `color` for simplicity. Moonstone Button uses plain `color` prop, so this could too I suppose.